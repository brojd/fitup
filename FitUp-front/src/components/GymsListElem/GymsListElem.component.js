import React, { Component, PropTypes } from 'react';
import Image from '../Image/Image.component';
import GymDetailsTd from '../GymDetailsTd/GymDetailsTd.component';
import styles from './GymsListElem.stylesheet.css';
import classNames from 'classnames';

class GymsListElem extends Component {
  constructor() {
    super();
    this._changePhoto = this._changePhoto.bind(this);
    this.state = {
      currentPhoto: ''
    };
  }
  _changePhoto(url) {
    this.setState({ currentPhoto: url });
  }
  componentDidMount() {
    this.setState({ currentPhoto: this.props.gym.photo_urls[0]});
  }
  
  render() {
    return (
      <table className={classNames('uk-width-10-10 uk-table', styles.gymTable)}>
        <thead>
          <tr>
            <th colSpan={5} className={classNames('uk-text-center', styles.gymTable__mainHeading)}>
              {this.props.gym.name}
            </th>
          </tr>
        </thead>
        <tbody>
          <tr>
            <td className={classNames('uk-width-3-10 uk-table-middle', styles.gymTable__td)}>
              <Image className={classNames('uk-thumbnail uk-align-center', styles.currentImg)}
                     alt={'gym logo'}
                     src={this.state.currentPhoto}
                     width={650} />
            </td>
            <GymDetailsTd gym={this.props.gym}/>
            <td className={classNames('uk-width-3-10', styles.gymTable__td)}>
              <h4 className={styles.gymTable__heading}>Description:</h4>
              <p className='uk-text uk-text-break uk-panel uk-panel-box'>
                {this.props.gym.description}
              </p>
            </td>
            <td className={classNames('uk-width-3-10', styles.gymTable__td)}>
              {this.props.gym.photo_urls.map((photoUrl, i) => {
                return (
                  <Image src={photoUrl}
                         alt={`${this.props.gym.name} photo`}
                         key={i}
                         className={classNames('uk-thumbnail uk-float-left', styles.thumbnail)}
                         width={108}
                         onClick={() => this._changePhoto(photoUrl)} />
                );
              })}
            </td>
          </tr>
        </tbody>
      </table>
    );
  }
}

GymsListElem.propTypes = {
  gym: PropTypes.object,
};

export default GymsListElem;
