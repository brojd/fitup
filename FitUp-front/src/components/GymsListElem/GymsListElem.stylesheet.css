.gymTable {
  background: linear-gradient(to top, #ffffff, rgba(136, 136, 136, 0.35));
  box-shadow: 8px 8px 6px rgba(202, 202, 202, 0.8), 0px 0px 10px #909090;
  margin-bottom: 50px !important;
  z-index: 1;
  border-radius: 4px;
  color: #4c4c4c;
  font-size: 13px; }

.gymTable__mainHeading {
  color: #ffffff;
  font-size: 16px;
  font-weight: 500;
  background: linear-gradient(to bottom, rgba(53, 0, 0, 0.81), #a70000);
  border-radius: 2px;
  text-transform: uppercase;
  letter-spacing: 3px;
  line-height: 14px; }

.gymTable__td {
  padding: 21px 21px 21px 0 !important; }

.gymTable__td:first-child {
  padding-left: 21px !important; }

.gymTable__heading {
  border-bottom: 1px solid rgba(80, 0, 0, 0.26); }

.thumbnail {
  cursor: pointer;
  margin-bottom: 14px !important; }

.thumbnail:nth-child(odd) {
  margin-right: 14px; }

/*# sourceMappingURL=GymsListElem.stylesheet.css.map */
