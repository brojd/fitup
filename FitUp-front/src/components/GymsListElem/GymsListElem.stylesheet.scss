@import "../../../sass_variables/colors";
@import "../../../sass_variables/widths";

.gymTable {
  background: linear-gradient(to top, #ffffff, rgba(136, 136, 136, 0.35));
  box-shadow: 8px 8px 6px rgba(202, 202, 202, 0.8), 0px 0px 10px #909090;
  margin-bottom: 50px !important;
  z-index: 1;
  border-radius: 4px;
  color: #4c4c4c;
  font-size: 13px;
}

.gymTable__mainHeading {
  color: #ffffff;
  font-size: 16px;
  font-weight: 500;
  background: linear-gradient(to bottom, rgba(53, 0, 0, 0.81), rgb(167, 0, 0));
  border-radius:2px;
  text-transform: uppercase;
  letter-spacing: 3px;
  line-height: 14px;
}

.gymTable__td {
  padding: 3*$whitespace 3*$whitespace 3*$whitespace 0 !important;
}

.gymTable__td:first-child {
  padding-left: 3*$whitespace !important;
}

.gymTable__heading {
  border-bottom: 1px solid rgba(80, 0, 0, 0.26);
}

.thumbnail {
  cursor: pointer;
  margin-bottom: 2*$whitespace !important;
}

.thumbnail:nth-child(odd) {
  margin-right: 2*$whitespace;
}
