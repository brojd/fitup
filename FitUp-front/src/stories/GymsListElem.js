import React from 'react';
import { storiesOf} from '@kadira/storybook';
import GymsListElem from '../components/GymsListElem/GymsListElem.component';
import GYMS from './mock_data/gyms';

storiesOf('GymsListElem', module)
  .add('GymsListElem component', () => (
    <GymsListElem key={GYMS[0].id} gym={GYMS[0]}/>
  ));
