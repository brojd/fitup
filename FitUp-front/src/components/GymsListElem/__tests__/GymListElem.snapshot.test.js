import React from 'react';
import GymsListElem from '../GymsListElem.component';
import renderer from 'react-test-renderer';

test('GymsListElem renders correctly', () => {

  const CURRENT_GYM = {
    '_id': {
      '$oid': '581ca32345633510184e0b5d'
    },
    'name': 'Fitness World',
    'address': {
      'city': 'Wroc≈Çaw',
      'postal_code': '50-001',
      'street': 'Olszowa',
      'number': 3
    },
    'description': 'Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.',
    'logo_url': 'http://www.feriogaj.com/wp-content/uploads/fitnes-world.png',
    'photo_urls': [
      'http://www.fru-fitness.pl/content/57/0/1370494719_6ca505f63f8064d96281ac502dd903f2.jpg',
      'http://www.top-gym.pl/files/dzialy/wydarzenia/lipiec%202015/Art%20Fitness2.png'
    ],
    'pricing': [
      {
        'name': '1 workout',
        'price': 15
      },
      {
        'name': '1 hour',
        'price': 40
      },
      {
        'name': '1 month',
        'price': 130
      }
    ],
    'cards_accepted': [
      {
        'name': 'Multisport',
        'accepted': false
      },
      {
        'name': 'ActiveSport',
        'accepted': false
      },
      {
        'name': 'OKsystem',
        'accepted': false
      },
      {
        'name': 'FitWro',
        'accepted': true
      },
      {
        'name': 'FitProfit',
        'accepted': false
      }
    ],
    'password': '$2a$10$ZqmPnzKm4QU5WRrZTkrhbup8UpDM55p2wreARh6cCc6FpUF3vv9Xa',
    'email': 'gym2@gym2.pl'
  };
  
  const component = renderer.create(
    <GymsListElem gym={CURRENT_GYM} />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
  
});
