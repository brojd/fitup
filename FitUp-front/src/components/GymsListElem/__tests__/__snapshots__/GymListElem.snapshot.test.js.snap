exports[`test GymsListElem renders correctly 1`] = `
<table
  className="uk-width-10-10 uk-table gymTable">
  <thead>
    <tr>
      <th
        className="uk-text-center gymTable__mainHeading"
        colSpan={5}>
        Fitness World
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <td
        className="uk-width-3-10 uk-table-middle gymTable__td">
        <img
          alt="gym logo"
          className="uk-thumbnail uk-align-center currentImg"
          height={150}
          onClick={undefined}
          src="http://www.fru-fitness.pl/content/57/0/1370494719_6ca505f63f8064d96281ac502dd903f2.jpg"
          width={650} />
      </td>
      <td
        className="uk-width-2-10 gymTable__td">
        <ul
          className="uk-list">
          <h4
            className="gymTable__heading">
            Fitness World
          </h4>
          <img
            alt="gym_logo"
            className="uk-float-right uk-width-3-10"
            height={150}
            onClick={undefined}
            src="http://www.feriogaj.com/wp-content/uploads/fitnes-world.png"
            width={150} />
          <li
            className="uk-width-4-10 uk-float-left">
            Olszowa
             
            3
            ,
          </li>
          <li
            className="uk-width-4-10 uk-float-left">
            Wroc≈Çaw
          </li>
          <div
            className="uk-clearfix" />
        </ul>
        <ul
          className="uk-list">
          <h4
            className="gymTable__heading">
            Pricing:
          </h4>
          <li>
            <div>
              1 workout
              : 
              15
               PLN
            </div>
            <div>
              1 hour
              : 
              40
               PLN
            </div>
            <div>
              1 month
              : 
              130
               PLN
            </div>
          </li>
        </ul>
        <ul
          className="uk-list">
          <h4
            className="gymTable__heading">
            Cards accepted:
          </h4>
          <li>
            <div
              className="uk-badge uk-margin-small-right uk-margin-small-bottom uk-width-3-10">
              FitWro
            </div>
          </li>
        </ul>
      </td>
      <td
        className="uk-width-3-10 gymTable__td">
        <h4
          className="gymTable__heading">
          Description:
        </h4>
        <p
          className="uk-text uk-text-break uk-panel uk-panel-box">
          Excepteur sint occaecat cupidatat non proident, sunt in culpa qui officia deserunt mollit anim id est laborum.
        </p>
      </td>
      <td
        className="uk-width-3-10 gymTable__td">
        <img
          alt="Fitness World photo"
          className="uk-thumbnail uk-float-left thumbnail"
          height={150}
          onClick={[Function]}
          src="http://www.fru-fitness.pl/content/57/0/1370494719_6ca505f63f8064d96281ac502dd903f2.jpg"
          width={108} />
        <img
          alt="Fitness World photo"
          className="uk-thumbnail uk-float-left thumbnail"
          height={150}
          onClick={[Function]}
          src="http://www.top-gym.pl/files/dzialy/wydarzenia/lipiec%202015/Art%20Fitness2.png"
          width={108} />
      </td>
    </tr>
  </tbody>
</table>
`;
